lib/src/rules/conformance/banned_property_write.dart

```diff
@@ -0,0 +1,240 @@
+    var leftHandSide = expression.leftHandSide;
+    DartType? dartTargetType;
+    late String dartProperty;
+    if (leftHandSide is PropertyAccess) {
+      dartTargetType = leftHandSide.realTarget.staticType;
+      dartProperty = leftHandSide.propertyName.name;
+    } else if (leftHandSide is PrefixedIdentifier) {
+      dartTargetType = leftHandSide.prefix.staticType;
+      dartProperty = leftHandSide.identifier.name;
+    } else {
+      return;
+    }
+
+    var dartTargetTypeElement = dartTargetType?.element;
+
+    if (dartTargetType == null || dartTargetTypeElement == null) return;
+
+    var dartHtmlBindingProperty = getWebLibraryMember(
+        nativeType: rule.nativeType, nativeMember: rule.nativeProperty);
+
+    var writeElement = expression.writeElement;
+    var isExternal = false;
+    if (writeElement is PropertyAccessorElement) {
+      // Use the variable declaration instead of the synthesized setter.
+      if (writeElement.isSynthetic) {
+        var variable = writeElement.variable;
+        if (variable is FieldElement) isExternal = variable.isExternal;
+      } else {
+        isExternal = writeElement.isExternal;
+      }
+    }
+
+    // If there is no `dart:html` binding for the given native type, we can use
+    // `@JS` or `@anonymous`. This means dynamic interop is possible.
+    var canUseDynamicInterop =
+        rule.nativeType != null && !hasDartHtmlBinding(rule.nativeType!);
+
+    // Report a lint if the target type is dynamic and one of the following
+    // holds true:
+    //
+    // 1. The property name matches the `dart:html` declaration name for the
+    // given native property.
+    // 2. The property name matches the given `dart:html` property.
+    // 3. The property name matches the given native property, and the type
+    // can be dynamically interop'd.
+    if (dartTargetType.isDynamic &&
+        (dartProperty == dartHtmlBindingProperty ||
+            dartProperty == rule.dartHtmlProperty ||
+            canUseDynamicInterop && dartProperty == rule.nativeProperty)) {
+      rule.reportLint(expression,
+          errorCode: _BannedPropertyWriteCode(
+              name: rule.name,
+              problemMessage:
+                  'This conformance check is being triggered because the '
+                  'static type of the target is dynamic.',
+              correctionMessage:
+                  'Try casting the target to a non-dynamic type.'));
+    } else if (dartTargetTypeElement is ClassElement) {
+      if (isExternal &&
+          dartTargetTypeElement.hasJS &&
+          dartProperty == rule.nativeProperty) {
+        // Only `@staticInterop` classes can interop the types bound to a
+        // `@Native` class. If there is no such binding however, then there
+        // needs to be a check for `@JS` and `@anonymous` classes as well.
+        if (isStaticInteropType(dartTargetTypeElement)) {
+          rule.reportLint(expression,
+              errorCode: _BannedPropertyWriteCode(
```

Comment:
I would recommend making instances of `_BannedPropertyWriteCode` be `static const` fields in one of these classes. It will not only ensure that the same instance is always being used, but will declutter the code.

Reply:
Hmmm, I like this idea, but I plan on adding a `BannedPropertyCode` that will refactor this code to a shared space (since the behavior will be very similar). The problem and correction messages will likely need to be modified based on things like whether the user called `getProperty` or `setProperty`. I could create `static const` messages for each variation, but that might duplicate code quite a bit. Thoughts?

Reply:
I understand the concern about duplicating code, and it's valid, but would there be any more duplication than there is today with the constructor invocations being inlined?

Anyway, it's not critical, and you're welcome to not do so now if you want to wait and see how it looks when you've made a base class.

Reply:
Once we check property gets, I believe so yes, since currently you're instantiating these errors as you need them, allowing you to modify the string values within the errors. With `static const` they'll take on fixed values before you can determine whether you want them for a `getProperty` or a `setProperty`.

That being said, maybe that's okay! I do agree it's much more readable that way and the benefit of doing so is that you have more flexibility in more descriptive errors. I'll try that out when I add code for property gets.

Reply:
I don't know whether this will address your concern / need, but the messages used to create error codes (which includes lint codes) can be parameterized. You could, for example, write the message to be something like `'... so this property {0} may violate conformance.'` and then
```
var readOrWrite = isForRead ? 'read' : 'write';
_rule.reportLint(expression, code, [readOrWrite]);
```

We generally don't use parameters for anything other than code snippets from the code being analyzed, but that has more to do with historical reasons and the fact that that often a desire to change a word like that means that we'd want to associate different fixes with the diagnostic than anything else.